src/lib/y2iscsi_client/finish_client.rb

```diff
@@ -0,0 +1,131 @@
+# |***************************************************************************
+# |
+# | Copyright (c) [2023] SUSE LLC
+# | All Rights Reserved.
+# |
+# | This program is free software; you can redistribute it and/or
+# | modify it under the terms of version 2 of the GNU General Public License as
+# | published by the Free Software Foundation.
+# |
+# | This program is distributed in the hope that it will be useful,
+# | but WITHOUT ANY WARRANTY; without even the implied warranty of
+# | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+# | GNU General Public License for more details.
+# |
+# | You should have received a copy of the GNU General Public License
+# | along with this program; if not, contact SUSE LLC
+# |
+# | To contact Novell about this file by physical or electronic mail,
+# | you may find current contact information at www.suse.com
+# |
+# |***************************************************************************
+
+require "yast"
+require "installation/finish_client"
+require "yast2/systemd/socket"
+require "yast2/execute"
+
+Yast.import "IscsiClientLib"
+Yast.import "Installation"
+Yast.import "Service"
+
+module Y2IscsiClient
+  # Finish client implementing the operations needed to transfer the iSCSI configuration to the
+  # target system at the end of installation
+  class FinishClient < ::Installation::FinishClient
+    include Yast::I18n
+
+    def initialize
+      super
+      textdomain "iscsi-client"
+    end
+
+    def title
+      _("Saving iSCSI configuration...")
+    end
+
+    def modes
+      [:installation, :update, :autoinst]
+    end
+
+    def write
+      copy_configuration
+      enable
+    end
+
+  private
+
+    # Copies open-iscsi configuration and databases to the target system
+    def copy_configuration
+      copy_directory("/etc/iscsi")
+      copy_directory("/var/lib/iscsi")
+    end
+
+    # Copies the content of the given directory to the target system only if the
+    # directory exists in the int-sys
+    #
+    # @param dir [String] path of the directory to copy
+    def copy_directory(dir)
+      return unless File.directory?(dir)
+
+      Yast::Execute.locally!(
+        "mkdir", "-p", File.join(Yast::Installation.destdir, dir), "&&",
```

Comment:
This doesn't work. I tried to keep it as close as possible to the original call to `WFM.Execute`. But `Yast::Execute` scapes the `&`s.

Reply:
The original code was:

```ruby
WFM.Execute(
          path(".local.bash"),
          "test -d /etc/iscsi/ && " \
            "mkdir -p #{Installation.destdir.shellescape}/etc/iscsi && " \
            "cp -a /etc/iscsi/* #{Installation.destdir.shellescape}/etc/iscsi/"
        )
)
```

Although all that could be done with pure Ruby, I wanted to keep it as close as possible to the original.

Reply:
hmm, how mkdir -p can fail? well, what you can do is in fact simple two Execute.locally!....when first failed, it will raise exception and skip the second one.